hpeOneView/resources/search/labels.py

```diff
@@ -37,46 +37,23 @@ class Labels(object):
     URI = '/rest/labels'
     RESOURCES_PATH = '/resources'
 
-    def __init__(self, con):
-        self._connection = con
-        self._client = ResourceClient(con, self.URI)
+    def __init__(self, connection, data=None):
+        super(Labels, self).__init__(connection, data)
 
-    def get_all(self, start=0, count=-1, filter='', sort=''):
+    def create(self, resource, timeout=-1):
         """
-        Gets a list of labels based on optional sorting and filtering and is constrained by start
-        and count parameters.
-
-        Args:
-            start:
-                The first item to return, using 0-based indexing.
-                If not specified, the default is 0 - start with the first available item.
-            count:
-                The number of resources to return. A count of -1 requests all items.
-                The actual number of items in the response might differ from the requested
-                count if the sum of start and count exceeds the total number of items.
-            filter (list or str):
-                A general filter/query string to narrow the list of items returned. The
-                default is no filter; all resources are returned.
-            sort:
-                The sort order of the returned data set. By default, the sort order is based
-                on create time with the oldest entry first.
-
-        Returns:
-            list: A list of labels.
-        """
-        return self._client.get_all(start=start, count=count, filter=filter, sort=sort)
-
-    def get(self, id_or_uri):
-        """
-        Gets a label by ID or URI.
+        Set all the labels for a resource.
 
         Args:
-            id_or_uri: Can be either the label ID or the label URI.
+            resource: The object containing the resource URI and a list of labels
 
         Returns:
-            dict: The label.
+            dict: Resource Labels
         """
-        return self._client.get(id_or_uri=id_or_uri)
+        uri = self.URI + self.RESOURCES_PATH
+        resource_data = self._helper.create(resource, uri=uri, timeout=timeout)
```

Comment:
instead of using the create method in resource_helper class, you can try using it from resource class itself as mentioned below.
```
return super(Labels, self).create(data, uri=uri, timeout=timeout)
```

Reply:
done